### PR TITLE
Bug 1856852 - Fix review checker not enough reviews card case

### PR DESCRIPTION
**Pull request by rahulsainani**
### What
– Update condition to display "not enough reviews card".
– It's displayed after retry succeeds, and rating, grade and highlights are all null.

### How
– Simplify middleware logic 
– Remove unused `AnalysisStatusDto` enum values
– Add Other `AnalysisStatusDto` value to differentiate between `error` and a different backend response.
– Update UI to have `NotEnoughReviews` as an error state

Note some logs were left accidentally in #4059 that are also being removed here.

### Preview
https://github.com/mozilla-mobile/firefox-android/assets/38040960/809ccc00-12fa-444c-b541-6114fdec8ef2

### Pull Request checklist

- [ ] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/changelog.md) or does not need one
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/shared/android/accessibility_guide.md) or does not include any user facing features

### After merge
- **Breaking Changes**: If this is a breaking Android Components change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### To download an APK when reviewing a PR (after all CI tasks finished running):
1. Click on `Checks` at the top of the PR page.
2. Click on the `firefoxci-taskcluster` group on the left to expand all tasks.
3. Click on the `build-apk-{fenix,focus,klar}-debug` task you're interested in.
4. Click on `View task in Taskcluster` in the new `DETAILS` section.
5. The APK links should be on the right side of the screen, named for each CPU architecture.

### GitHub Automation
https://bugzilla.mozilla.org/show_bug.cgi?id=1856852

### CODE REVIEW

File: fenix/app/src/main/java/org/mozilla/fenix/shopping/middleware/ReviewQualityCheckNetworkMiddleware.kt [hunk 1]
```diff
@@ -127,21 +126,9 @@ class ReviewQualityCheckNetworkMiddleware(
     private suspend fun pollForAnalysisStatus(): AnalysisStatusDto? =
         retry(
             predicate = { it == AnalysisStatusDto.PENDING || it == AnalysisStatusDto.IN_PROGRESS },
-            block = {
-                logger.debug("Retrying")
-                reviewQualityCheckService.analysisStatus()
-            },
+            block = { reviewQualityCheckService.analysisStatus() },
```
Comment (MozillaNoah) [inline fenix/app/src/main/java/org/mozilla/fenix/shopping/middleware/ReviewQualityCheckNetworkMiddleware.kt:129]: alternatively:
```suggestion
            block = reviewQualityCheckService::analysisStatus,
```

File: fenix/app/src/main/java/org/mozilla/fenix/shopping/middleware/ReviewQualityCheckService.kt [hunk 2]
```diff
@@ -100,7 +100,7 @@ class DefaultReviewQualityCheckService(
                 tab.engineState.engineSession?.requestAnalysisStatus(
                     url = tab.content.url,
                     onResult = {
-                        continuation.resume(it.asEnumOrDefault<AnalysisStatusDto>())
+                        continuation.resume(it.asEnumOrDefault(AnalysisStatusDto.OTHER))
```
Comment (MozillaNoah) [inline fenix/app/src/main/java/org/mozilla/fenix/shopping/middleware/ReviewQualityCheckService.kt:103]: non-blocking:
now that `asEnumOrDefault` is being passed a default value, should the function be simplified and made non-null and simply hardcode `OTHER` as the fallback?

File: fenix/app/src/test/java/org/mozilla/fenix/shopping/middleware/ProductAnalysisMapperTest.kt [hunk 3]
```diff
@@ -136,15 +136,45 @@ class ProductAnalysisMapperTest {
...
+        ).toProductReviewState(isInitialAnalysis = false)
+
+        val expected = ReviewQualityCheckState.OptedIn.ProductReviewState.Error.NotEnoughReviews
+
+        assertEquals(expected, actual)
+    }
+
+    @Test
+    fun `WHEN grade, rating and highlights are not all null and it is not initial analysis THEN it is mapped to AnalysisPresent`() {
+        val actual = ProductAnalysisTestData.productAnalysis(
```
Comment (MozillaNoah) [inline fenix/app/src/test/java/org/mozilla/fenix/shopping/middleware/ProductAnalysisMapperTest.kt:166]: would there be any value in adding coverage for the other (6?) happy path combinations?

**Review by MozillaNoah — APPROVED** [review]

File: fenix/app/src/main/java/org/mozilla/fenix/shopping/middleware/ReviewQualityCheckService.kt (hunk 2 above)
Comment (rahulsainani) [reply fenix/app/src/main/java/org/mozilla/fenix/shopping/middleware/ReviewQualityCheckService.kt:103]: Yeah that would make sense for `ReviewQualityCheckService`, but I think we should move this to common/middleware package as  product recommendations can use the same functionality and the existing String.toGrade mapping in `ProductAnalysisMapper.kt` can be replace by the common function, wdyt?

File: fenix/app/src/test/java/org/mozilla/fenix/shopping/middleware/ProductAnalysisMapperTest.kt (hunk 3 above)
Comment (rahulsainani) [reply fenix/app/src/test/java/org/mozilla/fenix/shopping/middleware/ProductAnalysisMapperTest.kt:166]: yeah, definitely doesn't hurt, added now, makes refactoring/updating/maintaining in the future easier.

File: fenix/app/src/main/java/org/mozilla/fenix/shopping/middleware/ReviewQualityCheckService.kt (hunk 2 above)
Comment (MozillaNoah) [reply fenix/app/src/main/java/org/mozilla/fenix/shopping/middleware/ReviewQualityCheckService.kt:103]: Makes sense to me

**Pull request merged**